### PR TITLE
Implement disk driver infrastructure

### PR DESCRIPTION
**Pull request by Jacobingalls**
We need a way to communicate with different storage controllers, just in case one is better than the other.

Drivers:
* `on-disk`: Creates volumes as folders. For testing only.
* `glusterfs`: **TODO** Connects to glusterfs

Example:
```
./run.sh -sc="on-disk" -scpath=(path, defaults to /etc/docker/mounts)
```
> Starts the plugin writing to the disk. By default the path is /etc/docker/mounts, but can be changed with `-scpath=/your/folder/here`.

**NOT PRODUCTION READY**

### CODE REVIEW

File: db/sqlite.go [hunk 1]
```diff
@@ -9,21 +9,23 @@ import (
...
-	var queries VolumeDatabaseQueries
-	queries.volumesCreateTableSQL = `CREATE TABLE IF NOT EXISTS volumes (
+// SQLiteSQLOverrides defines a list of qurries that need to differ from the defaults in order for sqlite to function correctly.
+var SQLiteSQLOverrides = VolumeDatabaseQueries{
+	volumesCreateTableSQL: `CREATE TABLE IF NOT EXISTS volumes (
         id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
         name TEXT NOT NULL UNIQUE,
         mountpoint TEXT
-    )`
+    )`,
```
Comment (JohnStarich) [inline db/sqlite.go]: Do we need a semicolon on the ends of these queries?

File: drivers/driver.go [hunk 2]
```diff
@@ -8,25 +8,28 @@ import (
 
 // RDMAVolumeDriver holds all the information pertaining to a RDMA Volume Driver.
 type RDMAVolumeDriver struct {
-	StorageController RDMAStorageController
+	StorageController StorageController
```
Comment (JohnStarich) [inline drivers/driver.go]: This syntax looks strange to me, how do Go structs work? My main confusion stems from seeing `StorageController` twice in a row.

File: main.go [hunk 3]
```diff
@@ -120,3 +134,37 @@ func validateDatabaseFlags(fatal bool, path bool, host bool, username bool, pass
...
+
+	default:
+		return nil, errors.New("Unsupported storage controller, please choose glusterfs or on-disk.")
+	}
+}
+
+func validateStorageFlags(fatal bool, path bool) {
+
+	var errors bool
+	noteErrorFunc := func(name string, value string, used bool) {
```
Comment (JohnStarich) [inline main.go]: Since we only invoke this anonymous function once, we should probably just write the logic in-line

File: drivers/driver.go (hunk 2 above)
Comment (Jacobingalls) [inline drivers/driver.go]: First column is the name, second is the type.

File: main.go (hunk 3 above)
Comment (Jacobingalls) [inline main.go]: When we get more storage flags, this will be expanded.

File: db/sqlite.go (hunk 1 above)
Comment (Jacobingalls) [inline db/sqlite.go]: Fixed in 03508fb

Comment (JohnStarich) [inline db/sqlite.go]: Looks like that's a different commit hash than you meant

**Pull request merged**